pandas/core/reshape/reshape.py

```diff
@@ -783,13 +788,25 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
     3  0  0
     4  0  0
 
+    >>> pd.get_dummies(pd.Series(list('abc')), dtype=float)
+         a    b    c
+    0  1.0  0.0  0.0
+    1  0.0  1.0  0.0
+    2  0.0  0.0  1.0
+
     See Also
     --------
     Series.str.get_dummies
     """
     from pandas.core.reshape.concat import concat
     from itertools import cycle
 
+    if dtype is None:
+        dtype = np.uint8
+
+    if np.dtype(dtype) is np.dtype('O'):
+        raise TypeError("'object' is not a valid type for get_dummies")
+
```

Comment:
this could be a ``ValueError``; also so ``dtype=object is not a valid dtype for get_dummies``